### DISCUSSION TITLE
Адрес по ФИАС

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#83] [Административная РГ]
Адрес по ФИАС затребован для Организации (устно высказанное пожелание МЗ РФ, для построения геоинформационных систем), и для пациента (лабораторная группа, дискуссия #54 )
На обсуждении 15.02.2022 высказаны следующие соображения:
- В.Родионов: возможно, достаточно одного кода адресного объекта по ФИАС, туда пишем код объекта нижнего уровня, остальные легко восстановить по справочнику (минусы - не очевидно, код какого объекта дан, поселок это или улица)
- В.Родионов: возможно, следует всю структуру ГАС оформить как Extension, забрать из справочника коды и наименования всех вышестоящих объектов по цепочке - удобнее иметь под рукой
- В.Родионов: если и включать структуры ФИАС, то только коды, а расшифровки (например "ул") не включать, потому что неудобно пользоваться при формировании текстового адреса
- А.Иванов: ФИАС (теперь ГАС) запаздывает, поэтому Хеликс, например, кодирует по ГАС до уровня улицы, а дом пишет числом.
- И.Осмоловский: СЭМДы принимают адрес по ФИАС до уровня адресного объекта - это все что до дома.

**Comment by eugenekogan-spb** [comment]
Отложили дискуссию, до момента, пока будут сформулированы вопросы.

**Comment by eugenekogan-spb** [comment]
Предложение (В.Родионов): 

Сделать специальный тип данных Address, он будет присутствовать в ресурсах Patient, Practitioner, Organisation. В нем сделать поля ФИАС как в CDA: код адресного объекта, наиболее детального до дома, и код дома.

Описание RuCore будет кроме профилей ресурсов включать также профили типов данных.

**Reply by Vrodvi** [reply]
Подготовлены макеты к данному предложению:

1) Определять в профилях ресурсов не базовый, а расширенный тип данных адреса:
![2022-03-03_13h03_48](https://user-images.githubusercontent.com/92928137/156542988-700d7653-3454-4e74-bcd0-b79c511b3990.png)

2) В расширенном типе данных адреса добавить расширение для кодирования по ФИАС:
![2022-03-03_13h04_17](https://user-images.githubusercontent.com/92928137/156543001-24a4c45c-6407-4034-8011-c85a585e5179.png)

3) Расширение ФИАС сделать с двумя атрибутами - указание GUID наиболее детального адресного объекта и указание GUID дома:
![2022-03-03_13h05_43](https://user-images.githubusercontent.com/92928137/156543011-13d42b2b-6abe-4df4-96cd-44888524aa58.png)

4) Пример ресурса пациент с адресом, расширенным кодированием по ФИАС:
![2022-03-03_13h06_24](https://user-images.githubusercontent.com/92928137/156543042-be885e07-2fd5-42ed-91f4-c47407d2800e.png)

**Reply by Vrodvi** [reply]
@eugenekogan-spb 
Евгений, уточните, пожалуйста - почему в итоге сделали только одно расширение с идентификатором до дома, куда делось расширение до адресного объекта?
Если смотреть по СЭМДам, то обязательным для заполнения как раз является идентификатор адресного объекта, а вот идентификатора дома может вообще не быть.

**Reply by eugenekogan-spb** [reply]
@Vrodvi похоже на пробу пера. Для полноценного расширения заготовлена страница docs/core/extensions/core-ex-address-fias.zd, однако она пока не наполнена. Думаю, опубликованные Вами в этой дискуссии макеты достаточны, осталось сделать страничку. Сделаем.

**Comment by GorbachevaT** [comment]
Добры день, коллеги! 
Хотелось бы уточнить вот такой вопрос: решено, что адрес по ФИАС будет в двух вариантах - "код, наиболее детальный до дома и код дома" это цитата. Иду на сайт ФИАС, ввожу адрес и понимаю что не понимаю. У любого адреса, если это не промышленное здание, три варианта:
1. (без квартиры и без дома) 
Город Санкт-Петербург, вн.тер.г. муниципальный округ Коломяги, ул Верхне-Каменская
**4d7c8a64-4a31-4e5d-a6dc-4139a63491f9**

2. ( без квартиры) 
Город Санкт-Петербург, вн.тер.г. муниципальный округ Коломяги, ул Верхне-Каменская, д. 3 к. 1 стр. 1
**267e1a5a-db27-400c-aaf4-503aa04366b9**

3. Полный
Город Санкт-Петербург, вн.тер.г. муниципальный округ Коломяги, ул Верхне-Каменская, д. 3 к. 1 стр. 1, кв. 84
**58917d62-7bfa-4a6b-b67f-a5708e01b9fd**

Почему все разговоры только про два варианта - без дома и с домом? А номер квартиры?

**Reply by Vrodvi** [reply]
# 1) По поводу выбора двух кодов для кодирования по ФИАС.
В ФИАС'е многое чего ещё есть, там и дальше квартиры есть деление - для указания части квартиры/комнат (для коммунальных квартир, например).
Также вообще не понятно, что означает кодирование адреса по ФИАС'у - какого-то единственного правильного варианта нет, поэтому нужно было принимать волевое решение.
При этом, хотя в ФИАС'е произвольная иерархия объектов  - от региона до частей квартиры, но технически там выделены конкретные артефакты - "адресный объект", затем "дом" и т.д.
Для принятия решения опирались на самые актуальные разработки в области российской информатизации в области здравоохранения - на структуру СЭМД'ов.
В СЭМД'ах кодирование адреса по ФИАС'у осуществляется с использованием двух атрибутов - кода "адресного объекта" и кода "дома". Т.е. использован подход с явным кодированием всех уровней иерархии по технической структуре ФИАС'а. При таком подходе в последствии можно добавлять следующие уровни уточняющих кодов далее по иерархии - код квартиры и т.д..

![image](https://github.com/fhir-ru/core/assets/92928137/d5b566d9-f9db-4bfd-bdac-18e0ce5b9385)
(пример из СЭМД "Выписка из протокола решения врачебной комиссии")



# 2) По поводу указания квартиры.
Тут интереснее. 

Во-первых, указание квартиры должно быть не в каждом адресе - у организаций их не будет, плюс частные дома. При этом желательно, чтобы структура адреса была единой. Возможно стоит обсудить формирование нескольких отдельных профилей адресов для разных "сущностей" (отдельно для физических лиц, отдельно для организаций/юридических лиц)? 

Во-вторых, важно различать указание адреса и его кодирование. Очевидно, что при необходимости передачи адреса он должен быть полным и сходиться с написанием в официальных документах (паспорте и т.п.) - в этом смысле указание квартиры для адреса пациента (в многоквартирном доме), к примеру, может являться обязательным. 
А вот каким образом закодировать этот же самый номер - уже не так очевидно, т.к. конкретных регламентов на этот счёт нет, а в официальных документах (паспорте, к примеру) никакого кодирования нет.

В СЭМД'ах эта проблема решается тем, что вся информация дублируется - есть человеко-читаемая часть и есть машино-обрабатываемая часть. Эти две части не должны противоречить друг другу, но не обязаны соответствовать на 100%. Т.е. в человеко-читаемой части может быть больше сведений чем в машино-обрабатываемой.
Вот к примеру указание квартиры в человеко-читаемой части:
![image](https://github.com/fhir-ru/core/assets/92928137/e1c807f2-87b9-4723-a944-3e536b80aadf)

Каким образом в профилях Ru Core учитывается наличие человеко-читаемой и машино-обрабатываемая частей информации - не понятно. Судя по всему изначально решили сделать вид, что такой проблемы нет, но вот пришло время прояснить этот момент - @eugenekogan-spb нужно где-то в документации описать подход сообщества Ru Core к проблеме ведения человеко-читаемой и машино-обрабатываемой информации.

В более узком смысле, проблема указания человеко-читаемой и машино-обрабатываемой частей информации конкретно для адреса должна была быть описана в соответствующем разделе документации, т.е. вот тут https://fhir-ru.zendoc.me/core.profile.Patient#:elements:address , но, увы, там сейчас присутствует лишь невнятная текстовка с неработающей ссылкой:
![image](https://github.com/fhir-ru/core/assets/92928137/6acb74e7-6c9f-4711-b1bf-70718ba4781b)
